File: spec/javascript/test_enum_of_value_inst.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('EnumOfValueInst', 'src/enum_0.bin', function(r, EnumOfValueInst) {
  assert.equal(r.pet1, EnumOfValueInst.Animal.CAT);
  assert.equal(r.pet2, EnumOfValueInst.Animal.CHICKEN);
  assert.equal(r.pet3, EnumOfValueInst.Animal.DOG);
  assert.equal(r.pet4, EnumOfValueInst.Animal.DOG);
});

File: spec/python/test_enum_of_value_inst.py
import unittest

from enum_of_value_inst import EnumOfValueInst

class TestEnumOfValueInst(unittest.TestCase):
    def test_enum_of_value_inst(self):
        r = EnumOfValueInst.from_file("src/enum_0.bin")

        self.assertEqual(r.pet_1, EnumOfValueInst.Animal.cat)
        self.assertEqual(r.pet_2, EnumOfValueInst.Animal.chicken)
        self.assertEqual(r.pet_3, EnumOfValueInst.Animal.dog)
        self.assertEqual(r.pet_4, EnumOfValueInst.Animal.dog)
